### PR TITLE
Issue #4632: override java parser report error to throw exception

### PR DESCRIPTION
**Pull request by rnveach**
Issue #4632

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/JavaParser.java [hunk 1]
```diff
@@ -85,13 +85,18 @@ public static DetailAST parse(FileContents contents)
         filter.hide(TokenTypes.SINGLE_LINE_COMMENT);
         filter.hide(TokenTypes.BLOCK_COMMENT_BEGIN);
 
-        final GeneratedJavaRecognizer parser = new GeneratedJavaRecognizer(filter);
+        final GeneratedJavaRecognizer parser = new GeneratedJavaRecognizer(filter) {
+            @Override
+            public void reportError(RecognitionException ex) {
+                throw new IllegalStateException(ex);
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/JavaParser.java]: `reportError` has no throws clause, so we can only throw runtime exceptions here.

**Comment by rnveach** [comment]
./.ci/shippable.sh no-exception-cassandra-storm-tapestry
> [ERROR] Failed to execute goal org.apache.maven.plugins:maven-site-plugin:3.3:site (default-site) on project sample: Error during page generation: Error rendering Maven report: Failed during checkstyle configuration: Exception was thrown while processing /root/src/github.com/checkstyle/checkstyle/.ci-temp/contribution/checkstyle-tester/src/main/java/tapestry-5/quickstart/filtered/archetype-resources/src/main/java/components/Layout.java: IllegalStateException occurred while parsing file /root/src/github.com/checkstyle/checkstyle/.ci-temp/contribution/checkstyle-tester/src/main/java/tapestry-5/quickstart/filtered/archetype-resources/src/main/java/components/Layout.java. /root/src/github.com/checkstyle/checkstyle/.ci-temp/contribution/checkstyle-tester/src/main/java/tapestry-5/quickstart/filtered/archetype-resources/src/main/java/components/Layout.java:1:10: expecting SEMI, found '{' -> [Help 1]

https://github.com/apache/tapestry-5/blob/master/quickstart/filtered/archetype-resources/src/main/java/components/Layout.java#L1
> package ${package}.components;

It is correct that this isn't a valid Java file.

It appears we never threw an exception on this before and because it is in the package, the parser kept on going. Here is an example printing a tree:
````
package ${package}.components;

public class TestClass {
    void method() {
    }
}
````

````
TestClass.java:1:10: expecting SEMI, found '{'
SEMI -> ; [1:29]
CLASS_DEF -> CLASS_DEF [3:0]
|--MODIFIERS -> MODIFIERS [3:0]
|  `--LITERAL_PUBLIC -> public [3:0]
|--LITERAL_CLASS -> class [3:7]
|--IDENT -> TestClass [3:13]
`--OBJBLOCK -> OBJBLOCK [3:23]
    |--LCURLY -> { [3:23]
    |--METHOD_DEF -> METHOD_DEF [4:4]
    |  |--MODIFIERS -> MODIFIERS [4:4]
    |  |--TYPE -> TYPE [4:4]
    |  |  `--LITERAL_VOID -> void [4:4]
    |  |--IDENT -> method [4:9]
    |  |--LPAREN -> ( [4:15]
    |  |--PARAMETERS -> PARAMETERS [4:16]
    |  |--RPAREN -> ) [4:16]
    |  `--SLIST -> { [4:18]
    |      `--RCURLY -> } [5:4]
    `--RCURLY -> } [6:0]
````

**Comment by rnveach** [comment]
I believe we just need to add file to ignore at https://github.com/checkstyle/contribution/blob/master/checkstyle-tester/projects-for-circle.properties#L16

**Comment by rnveach** [comment]
Regression: http://rveach.no-ip.org/checkstyle/regression/reports/233/hibernate-orm

hibernate-orm has a new exception when we were processing the file before.
http://rveach.no-ip.org/checkstyle/regression/reports/233/hibernate-orm/index.html#A7
http://rveach.no-ip.org/checkstyle/regression/reports/233/hibernate-orm/index.html#A122

No other major differences.

**Comment by romani** [comment]
Shippable failure:
```
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 11.954 s
[INFO] Finished at: 2019-03-10T19:30:59+00:00
[INFO] Final Memory: 29M/363M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-site-plugin:3.3:site (default-site) on project sample: Error during page generation: Error rendering Maven report: Failed during checkstyle configuration: Exception was thrown while processing /root/src/github.com/checkstyle/checkstyle/.ci-temp/contribution/checkstyle-tester/src/main/java/tapestry-5/quickstart/filtered/archetype-resources/src/main/java/components/Layout.java: IllegalStateException occurred while parsing file /root/src/github.com/checkstyle/checkstyle/.ci-temp/contribution/checkstyle-tester/src/main/java/tapestry-5/quickstart/filtered/archetype-resources/src/main/java/components/Layout.java. /root/src/github.com/checkstyle/checkstyle/.ci-temp/contribution/checkstyle-tester/src/main/java/tapestry-5/quickstart/filtered/archetype-resources/src/main/java/components/Layout.java:1:10: expecting SEMI, found '{' -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-site-plugin:3.3:site (default-site) on project sample: Error during page generation
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoExecutionException: Error during page generation
	at org.apache.maven.plugins.site.SiteMojo.execute(SiteMojo.java:143)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
	... 20 more
Caused by: org.apache.maven.doxia.siterenderer.RendererException: Error rendering Maven report: Failed during checkstyle configuration
	at org.apache.maven.plugins.site.ReportDocumentRenderer.renderDocument(ReportDocumentRenderer.java:233)
	at org.apache.maven.doxia.siterenderer.DefaultSiteRenderer.renderModule(DefaultSiteRenderer.java:319)
	at org.apache.maven.doxia.siterenderer.DefaultSiteRenderer.render(DefaultSiteRenderer.java:135)
	at org.apache.maven.plugins.site.SiteMojo.renderLocale(SiteMojo.java:175)
	at org.apache.maven.plugins.site.SiteMojo.execute(SiteMojo.java:138)
	... 22 more
Caused by: org.apache.maven.reporting.MavenReportException: Failed during checkstyle configuration
	at org.apache.maven.plugin.checkstyle.AbstractCheckstyleReport.executeReport(AbstractCheckstyleReport.java:488)
	at org.apache.maven.plugin.checkstyle.CheckstyleReport.executeReport(CheckstyleReport.java:154)
	at org.apache.maven.reporting.AbstractMavenReport.generate(AbstractMavenReport.java:255)
	at org.apache.maven.plugins.site.ReportDocumentRenderer.renderDocument(ReportDocumentRenderer.java:219)
	... 26 more
Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: Exception was thrown while processing /root/src/github.com/checkstyle/checkstyle/.ci-temp/contribution/checkstyle-tester/src/main/java/tapestry-5/quickstart/filtered/archetype-resources/src/main/java/components/Layout.java
	at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:304)
	at com.puppycrawl.tools.checkstyle.Checker.process(Checker.java:217)
	at org.apache.maven.plugin.checkstyle.exec.DefaultCheckstyleExecutor.executeCheckstyle(DefaultCheckstyleExecutor.java:252)
	at org.apache.maven.plugin.checkstyle.AbstractCheckstyleReport.executeReport(AbstractCheckstyleReport.java:473)
	... 29 more
Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: IllegalStateException occurred while parsing file /root/src/github.com/checkstyle/checkstyle/.ci-temp/contribution/checkstyle-tester/src/main/java/tapestry-5/quickstart/filtered/archetype-resources/src/main/java/components/Layout.java.
	at com.puppycrawl.tools.checkstyle.JavaParser.parse(JavaParser.java:103)
	at com.puppycrawl.tools.checkstyle.TreeWalker.processFiltered(TreeWalker.java:187)
	at com.puppycrawl.tools.checkstyle.api.AbstractFileSetCheck.process(AbstractFileSetCheck.java:81)
	at com.puppycrawl.tools.checkstyle.Checker.processFile(Checker.java:329)
	at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:291)
	... 32 more
Caused by: java.lang.IllegalStateException: /root/src/github.com/checkstyle/checkstyle/.ci-temp/contribution/checkstyle-tester/src/main/java/tapestry-5/quickstart/filtered/archetype-resources/src/main/java/components/Layout.java:1:10: expecting SEMI, found '{'
	at com.puppycrawl.tools.checkstyle.JavaParser$1.reportError(JavaParser.java:91)
	at com.puppycrawl.tools.checkstyle.grammar.GeneratedJavaRecognizer.packageDefinition(GeneratedJavaRecognizer.java:275)
	at com.puppycrawl.tools.checkstyle.grammar.GeneratedJavaRecognizer.compilationUnit(GeneratedJavaRecognizer.java:175)
	at com.puppycrawl.tools.checkstyle.JavaParser.parse(JavaParser.java:97)
	... 36 more
Caused by: /root/src/github.com/checkstyle/checkstyle/.ci-temp/contribution/checkstyle-tester/src/main/java/tapestry-5/quickstart/filtered/archetype-resources/src/main/java/components/Layout.java:1:10: expecting SEMI, found '{'
	at antlr.Parser.match(Parser.java:211)
	at com.puppycrawl.tools.checkstyle.grammar.GeneratedJavaRecognizer.packageDefinition(GeneratedJavaRecognizer.java:270)
	... 38 more
[ERROR] 
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
Caught: groovy.lang.GroovyRuntimeException: Error: !
groovy.lang.GroovyRuntimeException: Error: !
	at launch.executeCmd(launch.groovy:310)
	at launch.executeCmd(launch.groovy:304)
	at launch$executeCmd$10.callCurrent(Unknown Source)
	at launch.runMavenExecution(launch.groovy:231)
	at launch$_generateCheckstyleReport_closure2.doCall(launch.groovy:91)
	at launch.generateCheckstyleReport(launch.groovy:70)
	at launch$generateCheckstyleReport$1.callCurrent(Unknown Source)
	at launch.run(launch.groovy:10)
```

relaunched ...

**Comment by romani** [comment]
ok to merge ones CI is green and comment above is addressed.

**Comment by rnveach** [comment]
@romani CI can't be green until https://github.com/checkstyle/contribution/pull/349 is merged

**Comment by rnveach** [comment]
@romani 
> > overrode java parser report error to throw exception

> please fix typo in commit message.

What typo? If you mean `overrode` it is spelled correctly. It is the past tense of `override`.
https://www.google.com/search?client=firefox-b-1-d&q=overrode

**Comment by romani** [comment]
ok, hard to dispute here with you.
I thought that do comments in form of "this commit will (commit message)", so part tense should not be used.

**Comment by romani** [comment]
contribution fix is merged.

**Comment by rnveach** [comment]
repushed the commit to be present tense.

**Pull request merged**